tools/samtools.wdl

```diff
@@ -285,7 +294,14 @@ task subsample {
             n_cores=$(nproc)
         fi
 
-        read_count="$(samtools view --threads "$n_cores" -c ~{bam})"
+        read_count="$(samtools view \
+            --threads "$n_cores" \
+            -f ~{filter.include_if_all} \
+            -F ~{filter.exclude_if_any} \
+            --rf ~{filter.include_if_any} \
+            -G ~{filter.exclude_if_all} \
+            -c ~{bam} \
+        )"
 
         if [[ "$read_count" -gt "~{desired_reads}" ]]; then
             # the BAM has at least ~{desired_reads} reads, meaning we should
```

Comment:
I just pushed these changes and then immediately realized the logic was busted.

If a filter is enabled and the resulting count is smaller than `desired_count`, the unfiltered bam will be output.
I'm not entirely sure how this logic _should_ work, but I know it isn't that.

The most straightforward fix is to drop the `-c` flag here, pipe through `tee tmp.bam | wc -l` and then if `read_count < desired_reads` then `mv tmp.bam $bam` and output the filtered BAM. (That won't work bc the input BAM is read-only, but whatever you get the gist.)

My issue with this is that with an arbitrarily large `desired_reads` and an active filter, this task becomes `samtools view $filter > output.bam` with a whole lot of complicated stuff wrapped around it doing nothing. Which makes me ask the question, shouldn't we just have a `samtools view $filter > output.bam` task by itself? I think the answer is "yes" and that would be easy for me to add. But then that new task is overlapping with this task under certain criteria. But maybe that's ok?

OR. Maybe this task needs a rebrand. Spit balling: but call it `samtools.filter_and_subsample`. Expose all the `samtools view` options. Keep all our subsampling logic. But instead of writing this task with the assumption that the BAM should be subsampled, that's just an option.
Problem with that idea: it's freakin complicated. That's gonna be messy documentation. But IDK. I think it's the right call. Gonna take lunch and come back to this problem.